RubiksCubeGAP.ipynb

```diff
@@ -0,0 +1,984 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 35,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "true"
+      ]
+     },
+     "execution_count": 35,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "G.in(@gap(\"(17,19)(11,8)(6,25)(7,28)(18,21)\"), cube)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 36,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "GAP: Group([ (2,34)(4,10)(5,26)(7,18)(12,37)(13,20)(15,44)(21,28)(23,42)(29,36)\n",
+       "  (31,45)(39,47) ])"
+      ]
+     },
+     "execution_count": 36,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "z = G.Centre(cube)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 37,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "GAP: <free group on the generators [ t, l, f, r, e, b ]>"
+      ]
+     },
+     "execution_count": 37,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "f = @gap(\"FreeGroup(\\\"t\\\", \\\"l\\\", \\\"f\\\", \\\"r\\\", \\\"e\\\", \\\"b\\\")\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "GAP: [ t, l, f, r, e, b ] -> \n",
+       "[ (1,3,8,6)(2,5,7,4)(9,33,25,17)(10,34,26,18)(11,35,27,19), \n",
+       "  (1,17,41,40)(4,20,44,37)(6,22,46,35)(9,11,16,14)(10,13,15,12), \n",
+       "  (6,25,43,16)(7,28,42,13)(8,30,41,11)(17,19,24,22)(18,21,23,20), \n",
+       "  (3,38,43,19)(5,36,45,21)(8,33,48,24)(25,27,32,30)(26,29,31,28), \n",
+       "  (1,14,48,27)(2,12,47,29)(3,9,46,32)(33,35,40,38)(34,37,39,36), \n",
+       "  (14,22,30,38)(15,23,31,39)(16,24,32,40)(41,43,48,46)(42,45,47,44) ]"
+      ]
+     },
+     "execution_count": 38,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "hom = G.GroupHomomorphismByImages( f, cube, G.GeneratorsOfGroup(f), G.GeneratorsOfGroup(cube))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 39,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "GAP: l*f*t*f^-1*t^-1*l^-1*f*t*f^2*l*f*l^-2*t*l*f^-1*t^-1*l^-1*t*l*f^2*t*l*t*l^-1*f^\\\n",
+       "-1*l*t^-1*l^-1*f*r*t^-1*r^-1*f^-1*l*t*f*t^-1*f^-1*l^-1*t*f^-1*l^-1*t^-1*l*t*f*\\\n",
+       "t^-2*f*t*f^-1*t^-1*l^-1*t^-1*l*t^-1*l^-1*t*e^-1*t*e*(l*t)^2*e*l^-1*e^-1*t^-1*l\\\n",
+       "^-3*b*f*b^-1*l^-1*f^-1*t*l^-1*f*t*f*l^-1*t^-1*b*r^-1*b^-1*t^-2*e^-1*r*e*r*f^-1\\\n",
+       "*e*t^-1*e^-1*r^-2*t^-2*l^-1*b^-1*r^-1*e^-1"
+      ]
+     },
+     "execution_count": 39,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "pre1 = G.PreImagesRepresentative(hom, GAP.call_gap_func(@gap(\"x -> x.1\"), z) )"
```

Comment:
For the moment, `G.PreImagesRepresentative(hom, G.GeneratorsOfGroup(z)[1])` would look more natural.
The GAP syntax `z.1` belongs to a special method only for groups, but if it would be regarded as interesting, one could make it available also in Julia. (I would not be in favor of this.)

Reply:
Hmm. I've figured out in the meantime that GAP functions can actually be called directly, e.g `@gap("x -> x + 1")(z)`; this might be illustrative, but I think your approach is more idiomatic. I'll think about that.

Reply:
You can also just write `pre1 = G.PreImagesRepresentative(hom, z.:1)`